hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -129,6 +129,9 @@ public static ClientMessage create() {
 
     public static ClientMessage createForEncode(int initialCapacity) {
         initialCapacity = QuickMath.nextPowerOfTwo(initialCapacity);
+        if (initialCapacity < 0) {
+            initialCapacity = Integer.MAX_VALUE;
+        }
         if (USE_UNSAFE) {
```

Comment:
QuickMath.nextPowerOfTwo are being called from places such as Int2ObjectHashMap. We may need to analyse and understand what would be side affects of changing this method. I would prefer not touching this method now as it is a bug fix period but surely we can open an issue for examining this change for later releases.